### PR TITLE
Some more backend work

### LINKED ISSUE #31: Content-Based Model

### LINKED ISSUE #71: Corequisite typo'd as "corerequisite"

### LINKED ISSUE #73: new get course detail api params pattern change discussion
may change from `ece220` to `ece/220`, 
to solve the issue for `mec e250A `and change to `mec_e/250A`

### PR DESCRIPTION
**Pull request by aelgohar**
- Changed API endpoint from ece493 to ece/493. Any spacing in course name will be represented with an "_" and no spacing in course number. So "MEC E 230B" becomes "mec_e/230b" @MXSH-Dev @raysarinas I think you'll need to do some changes on the frontend here
- Added secrets file in config directory, contains the postgres remote url
- Fixed migrations and pushed data to remote db
- Added multiprocessing to the scraping, it now takes 10% of the original time. 
- Finished content based model
- Added similar courses to the course endpoint
- Fixed "core requisite" bug
closes #31 
closes #71 
closes #73

### CODE REVIEW

File: server/models/course.py [hunk 1]
```diff
@@ -17,11 +17,7 @@ class Course(db.Model):
 
     prerequisites = db.Column(db.Text())
     antirequisites = db.Column(db.Text())
-    corerequisites = db.Column(db.Text())
-
-    __table_args__ = (
-        db.UniqueConstraint("subject_code", "number", name="full_course_name"),
-    )
```
Comment (raysarinas) [inline server/models/course.py:24]: is there a reason you removed this? looks like you added something similar to the `instructor` model tho

**Review by raysarinas — APPROVED** [review]
LGTM, just made one comment about the `course` model.

File: server/models/course.py (hunk 1 above)
Comment (aelgohar) [reply server/models/course.py:24]: Nope, I had a bug in the code where some courses were getting scraped twice so I thought this didn't make sense. 

I fixed the bug so I will add this back.

Comment (raysarinas) [reply server/models/course.py:24]: Sounds good, I think the scraping twice thing is a problem with catalogue (seems like some courses actually are listed twice)

**Pull request merged**